wdl-analysis/src/diagnostics.rs

```diff
@@ -792,3 +792,11 @@ pub fn invalid_placeholder_option<N: TreeNode>(
 
     Diagnostic::error(message).with_label(format!("this is type `{ty}`"), span)
 }
+
+/// Creates an invalid regex pattern diagnostic.
+pub fn invalid_regex_pattern(function: &str, pattern: &str, error: &str, span: Span) -> Diagnostic {
+    Diagnostic::error(format!(
+        "invalid regex pattern `{function}`: `{pattern}` is not a valid regular expression"
```

Comment:
This will require updating the test baseline:

```suggestion
        "invalid regular expression `{pattern}` used in function `{function}`: {error}"
```